### ISSUE TITLE
TPH + OwnsOne(CustomCollection<T>) = Extra table for each target type

### ISSUE THREAD

**Issue by aureole82** [dotnet/efcore#19543]
[EF.TPHAndOwnsOne.zip](https://github.com/dotnet/efcore/files/4041785/EF.TPHAndOwnsOne.zip)
```cs
public abstract class Process
{
    public int Id { get; set; }
    public SerializableCollection<string> Notes { get; set; } = new SerializableCollection<string>();
}
public class Application : Process
{
    public SerializableCollection<Skill> Skills { get; set; } = new SerializableCollection<Skill>();
}
public class Employment : Process
{
}
public class MyDbContext : DbContext
{
    public DbSet<Application> Applications { get; set; }
    public DbSet<Process> Processes { get; set; }
    public DbSet<Employment> Employments { get; set; }

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.Entity<Process>(entity => entity.OwnsOne(process => process.Notes));
        modelBuilder.Entity<Application>(entity => entity.OwnsOne(application => application.Skills));
    }
}
```
### What's wrong
```ps
PM> Add-Migration Initial
```
creates 
* `Processes` table 
* `Processes1` table for `Application.Skills`
* `Processes2` table for `Process.Notes`

### What's expected
It's default that ALL properties of owned types should be nested inside the owner table.

Please note: It worked in EntityFramework Core version: 2.2!

Please note (2): Using `public SerializableCollection<string> Skills { get; set; } = new SerializableCollection<string>();` inside `Application` class `Add-Migration` creates a decent schema.

### Steps to reproduce
See attached simplified project.

### Further technical details

EF Core version: 3.1.0
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: NET Core 3.1
Operating system: Windows 10
IDE: Visual Studio 2019 Version 16.4.2

**Comment by aureole82** [comment]
Explicitly naming the owner table might be a dirty workaround:
```cs
modelBuilder.Entity<Process>(entity => entity
    .OwnsOne(process => process.Notes)
    .ToTable(nameof(Processes))
);
modelBuilder.Entity<Application>(entity => entity
    .OwnsOne(application => application.Skills)
    .ToTable(nameof(Processes))
);
```

Is there a better way than using `nameof(Processes)`. It will break if someone renames the `Processes` table by calling `ToTable("ProcessesRenamed")`.

**Comment by AndriySvyryd** [comment]
Fixed in fb33fe4b54771afd2709cabd76bc585bd9913fbe
